weaviate/collections/classes/grpc.py

```diff
@@ -228,11 +228,38 @@ class Rerank(_WeaviateInput):
     query: Optional[str] = Field(default=None)
 
 
+class _MultiVectorQuery(_WeaviateInput):
+    tensor: Sequence[Sequence[float]]
+
+
+class _ManyVectorsQuery(_WeaviateInput):
+    vectors: Sequence[Sequence[float]]
+
+
 NearVectorInputType = Union[
-    Sequence[NUMBER], Dict[str, Union[Sequence[NUMBER], Sequence[Sequence[NUMBER]]]]
+    Sequence[NUMBER],
+    Sequence[Sequence[NUMBER]],
+    Mapping[
+        str,
+        Union[Sequence[NUMBER], Sequence[Sequence[NUMBER]], _MultiVectorQuery, _ManyVectorsQuery],
+    ],
 ]
 
 
+class NearVector:
+    """Factory class to use when defining near vector queries with multiple vectors in `near_vector()` and `hybrid()` methods."""
+
+    @staticmethod
+    def multidimensional(tensor: Sequence[Sequence[float]]) -> _MultiVectorQuery:
+        """Define a multi-vector query to be used within a near vector search, i.e. a single vector over a multi-vector space."""
+        return _MultiVectorQuery(tensor=tensor)
+
+    @staticmethod
+    def list_of_vectors(vectors: Sequence[Sequence[float]]) -> _ManyVectorsQuery:
```

Comment:
these should be added to output :)

Reply:
They're already in `weaviate/classes/query.py` since they're required when inputting information from the user. Data returned from Weaviate will still be of type `Sequence[Sequence[float]]` without the need to differentiate between the two cases